File: CS/ChartWithPivotGrid/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ChartWithPivotGrid
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nwindDataSet.SalesPerson' table. You can move, or remove it, as needed.
            this.salesPersonTableAdapter.Fill(this.nwindDataSet.SalesPerson);
            chartControl1.DataSource = this.pivotGridControl1;
            chartControl1.SeriesDataMember = "Series";
            chartControl1.SeriesTemplate.ArgumentDataMember = "Arguments";
            chartControl1.SeriesTemplate.ValueDataMembers.AddRange(new string[] { "Values" });
        }

        private void pivotGridControl1_FieldAreaChanged(object sender, DevExpress.XtraPivotGrid.PivotFieldEventArgs e)
        {
            if (fieldOrderDate.Area == DevExpress.XtraPivotGrid.PivotArea.DataArea)
            {
                chartControl1.SeriesTemplate.ValueScaleType = DevExpress.XtraCharts.ScaleType.DateTime;
                chartControl1.DataSource = pivotGridControl1;
            }
            if (fieldQuantity.Area == DevExpress.XtraPivotGrid.PivotArea.DataArea)
            {
                chartControl1.SeriesTemplate.ValueScaleType = DevExpress.XtraCharts.ScaleType.Numerical;
                chartControl1.DataSource = pivotGridControl1;
            }
        }

        private void pivotGridControl1_FieldAreaChanging(object sender, DevExpress.XtraPivotGrid.PivotAreaChangingEventArgs e)
        {
            chartControl1.DataSource = null;
            if (e.Field == fieldOrderDate && e.NewArea == DevExpress.XtraPivotGrid.PivotArea.DataArea && fieldQuantity.Area == DevExpress.XtraPivotGrid.PivotArea.DataArea)
                e.Allow = false;
            if (e.Field == fieldQuantity && e.NewArea == DevExpress.XtraPivotGrid.PivotArea.DataArea && fieldOrderDate.Area == DevExpress.XtraPivotGrid.PivotArea.DataArea)
                e.Allow = false; 
        }
    }
}

File: VB/ChartWithPivotGrid/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace ChartWithPivotGrid
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.SalesPerson' table. You can move, or remove it, as needed.
			Me.salesPersonTableAdapter.Fill(Me.nwindDataSet.SalesPerson)
			chartControl1.DataSource = Me.pivotGridControl1
			chartControl1.SeriesDataMember = "Series"
			chartControl1.SeriesTemplate.ArgumentDataMember = "Arguments"
			chartControl1.SeriesTemplate.ValueDataMembers.AddRange(New String() { "Values" })
		End Sub

		Private Sub pivotGridControl1_FieldAreaChanged(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotFieldEventArgs) Handles pivotGridControl1.FieldAreaChanged
			If fieldOrderDate.Area = DevExpress.XtraPivotGrid.PivotArea.DataArea Then
				chartControl1.SeriesTemplate.ValueScaleType = DevExpress.XtraCharts.ScaleType.DateTime
				chartControl1.DataSource = pivotGridControl1
			End If
			If fieldQuantity.Area = DevExpress.XtraPivotGrid.PivotArea.DataArea Then
				chartControl1.SeriesTemplate.ValueScaleType = DevExpress.XtraCharts.ScaleType.Numerical
				chartControl1.DataSource = pivotGridControl1
			End If
		End Sub

		Private Sub pivotGridControl1_FieldAreaChanging(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotAreaChangingEventArgs) Handles pivotGridControl1.FieldAreaChanging
			chartControl1.DataSource = Nothing
			If e.Field Is fieldOrderDate AndAlso e.NewArea = DevExpress.XtraPivotGrid.PivotArea.DataArea AndAlso fieldQuantity.Area = DevExpress.XtraPivotGrid.PivotArea.DataArea Then
				e.Allow = False
			End If
			If e.Field Is fieldQuantity AndAlso e.NewArea = DevExpress.XtraPivotGrid.PivotArea.DataArea AndAlso fieldOrderDate.Area = DevExpress.XtraPivotGrid.PivotArea.DataArea Then
				e.Allow = False
			End If
		End Sub
	End Class
End Namespace